src/main/kotlin/com/openlattice/postgres/PostgresDataTables.kt

```diff
@@ -139,6 +141,10 @@ class PostgresDataTables {
                     .ifNotExists()
                     .desc()
 
+            val originIdNotNullIndex = PostgresExpressionIndexDefinition(tableDefinition, "(${ORIGIN_ID.name} != '${IdConstants.EMPTY_ORIGIN_ID.id}')" )
```

Comment:
I don't think it's safe to use a constant for an empty origin id. For space reasons and for avoiding collisions with actual origin ids drawn form sequential uuid space, we  should use IS NOT NULL.

Reply:
`ORIGIN_ID` needs to be part of the PrimaryKey so it can't be null :(